discussions-store/discussions-store-default/src/main/java/org/xwiki/contrib/discussions/store/internal/DefaultMessageHolderReferenceService.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions.store.internal;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpSession;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.discussions.DiscussionStoreConfigurationParameters;
+import org.xwiki.contrib.discussions.domain.references.DiscussionReference;
+import org.xwiki.contrib.discussions.store.MessageHolderReferenceService;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Default implementation of {@link MessageHolderReferenceService}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@Component
+@Singleton
+public class DefaultMessageHolderReferenceService implements MessageHolderReferenceService
+{
+    static final String MESSAGE_HOLDER_REFERENCE_SESSION_ATTRIBUTE = "xwikiDiscussionsMessageHolderReferences";
```

Comment:
no visibility?

Reply:
indeed, I could keep it strictly private